src/nnue/layers/affine_transform.h

```diff
@@ -177,6 +186,55 @@ namespace Eval::NNUE::Layers {
         sum = _mm_hadd_epi32(sum, sum);
         output[i] = _mm_cvtsi128_si32(sum);
 
+  #elif defined(USE_SSE2)
+        __m128i sum_lo = _mm_cvtsi32_si128(biases_[i]);
+        __m128i sum_hi = kZeros;
+        const auto row = reinterpret_cast<const __m128i*>(&weights_[offset]);
+        for (IndexType j = 0; j < kNumChunks; ++j) {
+          __m128i row_j = _mm_load_si128(&row[j]);
+          __m128i input_j = _mm_load_si128(&input_vector[j]);
+          __m128i row_signs = _mm_cmpgt_epi8(kZeros, row_j);
+          __m128i extended_row_lo = _mm_unpacklo_epi8(row_j, row_signs);
+          __m128i extended_row_hi = _mm_unpackhi_epi8(row_j, row_signs);
+          __m128i extended_input_lo = _mm_unpacklo_epi8(input_j, kZeros);
+          __m128i extended_input_hi = _mm_unpackhi_epi8(input_j, kZeros);
+          __m128i product_lo = _mm_mullo_epi16(extended_row_lo, extended_input_lo);
+          __m128i product_hi = _mm_mullo_epi16(extended_row_hi, extended_input_hi);
+          product_lo = _mm_madd_epi16(product_lo, kOnes);
+          product_hi = _mm_madd_epi16(product_hi, kOnes);
```

Comment:
Wat are you doing @Fanael ! Stahp ! The old CPU don't have many multipliers. Just do
 ```
product_lo = _mm_madd_epi16(extended_row_lo, extended_input_lo);
product_hi = _mm_madd_epi16(extended_row_hi, extended_input_hi);
```
like I did. It's guaranteed not to overflow.

Reply:
Good catch, thanks!